### PR TITLE
Refactor cql command

### PR DESCRIPTION
**Pull request by laodouya**
Refactor cql command into subcommands.

Now has adapter balance console create drop help permission transfer web.

### CODE REVIEW

File: cmd/cql/internal/adapter.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	"github.com/CovenantSQL/CovenantSQL/utils"
+)
+
+var (
+	adapterAddr string // adapter listen addr
+
+	adapterHTTPServer *http.Server
+)
+
+var CmdAdapter = &Command{
```
Comment (covenantdb) [inline cmd/cql/internal/adapter.go:35]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported var CmdAdapter should have comment or be unexported

File: cmd/cql/internal/balance.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+	"github.com/CovenantSQL/CovenantSQL/client"
+	"github.com/CovenantSQL/CovenantSQL/types"
+)
+
+var (
+	tokenName string // get specific token's balance of current account
+)
+
+var CmdBalance = &Command{
```
Comment (covenantdb) [inline cmd/cql/internal/balance.go:31]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported var CmdBalance should have comment or be unexported

File: cmd/cql/internal/base.go [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+// Name returns the command's short name: the last word in the usage line before a flag or argument.
+func (c *Command) Name() string {
+	name := c.LongName()
+	if i := strings.LastIndex(name, " "); i >= 0 {
+		name = name[i+1:]
+	}
+	return name
+}
+
+func (c *Command) Usage() {
```
Comment (covenantdb) [inline cmd/cql/internal/base.go:87]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported method Command.Usage should have comment or be unexported

File: cmd/cql/internal/base.go [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+// Runnable reports whether the command can be run; otherwise
+// it is a documentation pseudo-command such as importpath.
+func (c *Command) Runnable() bool {
+	return c.Run != nil
+}
+
+var atExitFuncs []func()
+
+func AtExit(f func()) {
```
Comment (covenantdb) [inline cmd/cql/internal/base.go]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported function AtExit should have comment or be unexported

File: cmd/cql/internal/base.go [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+	atExitFuncs = append(atExitFuncs, f)
+}
+
+func Exit() {
```
Comment (covenantdb) [inline cmd/cql/internal/base.go:106]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported function Exit should have comment or be unexported

File: cmd/cql/internal/base.go [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+	for _, f := range atExitFuncs {
+		f()
+	}
+	os.Exit(exitStatus)
+}
+
+func ExitIfErrors() {
```
Comment (covenantdb) [inline cmd/cql/internal/base.go:114]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported function ExitIfErrors should have comment or be unexported

File: cmd/cql/internal/base.go [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+	if exitStatus != 0 {
+		Exit()
+	}
+}
+
+var exitStatus = 0
+var exitMu sync.Mutex
+
+func SetExitStatus(n int) {
```
Comment (covenantdb) [inline cmd/cql/internal/base.go:124]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported function SetExitStatus should have comment or be unexported

File: cmd/cql/internal/console.go [hunk 8]
```diff
@@ -0,0 +1,388 @@
...
+	"github.com/xo/usql/env"
+	"github.com/xo/usql/handler"
+	"github.com/xo/usql/rline"
+	"github.com/xo/usql/text"
+
+	"github.com/CovenantSQL/CovenantSQL/client"
+	"github.com/CovenantSQL/CovenantSQL/utils"
+)
+
+var CmdConsole = &Command{
```
Comment (covenantdb) [inline cmd/cql/internal/console.go:48]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported var CmdConsole should have comment or be unexported

File: cmd/cql/internal/create.go [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strconv"
+
+	"github.com/CovenantSQL/CovenantSQL/client"
+)
+
+var CmdCreate = &Command{
```
Comment (covenantdb) [inline cmd/cql/internal/create.go:28]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported var CmdCreate should have comment or be unexported

File: cmd/cql/internal/drop.go [hunk 10]
```diff
@@ -0,0 +1,57 @@
+package internal
+
+import (
+	"github.com/CovenantSQL/CovenantSQL/client"
+)
+
+var CmdDrop = &Command{
```
Comment (covenantdb) [inline cmd/cql/internal/drop.go:24]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported var CmdDrop should have comment or be unexported

File: cmd/cql/internal/help.go [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+)
+
+const name = "cql"
+
+var (
+	// Version of command, set by main func of version
+	Version = "unknown"
+)
+
+var CmdVersion = &Command{
```
Comment (covenantdb) [inline cmd/cql/internal/help.go:33]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported var CmdVersion should have comment or be unexported

File: cmd/cql/internal/help.go [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+	UsageLine:   "cql version",
+	Description: "Show cql build version infomation",
+}
+
+func init() {
+	CmdVersion.Run = runVersion
+}
+
+func PrintVersion(printLog bool) string {
```
Comment (covenantdb) [inline cmd/cql/internal/help.go:56]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported function PrintVersion should have comment or be unexported

File: cmd/cql/internal/permission.go [hunk 13]
```diff
@@ -0,0 +1,93 @@
...
+
+import (
+	"encoding/json"
+
+	"github.com/CovenantSQL/CovenantSQL/client"
+	"github.com/CovenantSQL/CovenantSQL/proto"
+	"github.com/CovenantSQL/CovenantSQL/types"
+)
+
+var CmdPermission = &Command{
```
Comment (covenantdb) [inline cmd/cql/internal/permission.go]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported var CmdPermission should have comment or be unexported

File: cmd/cql/internal/transfer.go [hunk 14]
```diff
@@ -0,0 +1,95 @@
...
+	"regexp"
+	"strconv"
+
+	"github.com/CovenantSQL/CovenantSQL/client"
+	"github.com/CovenantSQL/CovenantSQL/crypto/hash"
+	"github.com/CovenantSQL/CovenantSQL/proto"
+	"github.com/CovenantSQL/CovenantSQL/types"
+)
+
+var CmdTransfer = &Command{
```
Comment (covenantdb) [inline cmd/cql/internal/transfer.go:31]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported var CmdTransfer should have comment or be unexported

File: cmd/cql/internal/web.go [hunk 15]
```diff
@@ -0,0 +1,63 @@
...
+)
+
+var (
+	webAddr string // Web addr
+
+	webService    *observer.Service
+	webHTTPServer *http.Server
+)
+
+var CmdWeb = &Command{
```
Comment (covenantdb) [inline cmd/cql/internal/web.go]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported var CmdWeb should have comment or be unexported

File: cmd/cql/internal/adapter.go [hunk 16]
```diff
@@ -0,0 +1,72 @@
+package internal
```
Comment (leventeliu) [inline cmd/cql/internal/adapter.go:17]: Should provide license message.

**Review by leventeliu — APPROVED** [review]
Great job, approve.

File: cmd/cql/internal/console.go [hunk 17]
```diff
@@ -46,8 +46,19 @@ import (
 
 // CmdConsole is cql console command entity.
 var CmdConsole = &Command{
-	UsageLine:   "cql console [-dsn dsn_string] [-command sqlcommand] [-file filename] [-out outputfile] [-no-rc true/false] [-single-transaction] [-variable variables]",
-	Description: "run a console for realtime sql operation",
+	UsageLine: "cql console [-config file] [-password masterkey] [-dsn dsn_string] [-command sqlcommand] [-file filename] [-out outputfile] [-no-rc true/false] [-single-transaction] [-variable variables] [-web web_addr] [-adapter adapter_addr]",
+	Short:     "run a console for realtime sql operation",
```
Comment (auxten) [inline cmd/cql/internal/console.go]: realtime -> interactive

File: cmd/cql/internal/web.go [hunk 18]
```diff
@@ -16,8 +32,13 @@ var (
 
 // CmdWeb is cql web command.
 var CmdWeb = &Command{
-	UsageLine:   "cql web [-tmp-path path] [-bg-log-level level] [address]",
-	Description: "Web command serve a database chain explorer, e.g. :8546",
+	UsageLine: "cql web [-config file] [-tmp-path path] [-bg-log-level level] [address]",
+	Short:     "start a database chain web explorer",
```
Comment (auxten) [inline cmd/cql/internal/web.go]: database chain -> SQLChain

File: cmd/cql/internal/console.go [hunk 19]
```diff
@@ -46,8 +46,19 @@ import (
...
+	Long: `
+Console command can run a realtime sql console for CovenantSQL
+The -dsn param is required
+e.g.
+	cql console -dsn covenant://the_dsn_of_your_database
```
Comment (auxten) [inline cmd/cql/internal/console.go]: no such covenant://

Comment (auxten) [inline cmd/cql/internal/console.go]: no "INFO[0000]" on cql help xxx

**Review by auxten — APPROVED** [review]
Nice work!

**Pull request merged**